packages/dd-trace/src/encode/notepack.js

```diff
@@ -0,0 +1,346 @@
+        size = 5;
+      } else {
+        throw new Error('String too long');
+      }
+      defers.push({ str: value, length: length, offset: bytes.length });
+      return size + length;
+    case 'number':
+      // TODO: encode to float 32?
+
+      if (Math.floor(value) !== value || !isFinite(value)) { // float 64
+        bytes.push(0xcb);
+        defers.push({ float: value, length: 8, offset: bytes.length });
+        return 9;
+      }
+
+      if (value >= 0) {
+        if (value < 0x80) { // positive fixnum
+          bytes.push(value);
+          return 1;
+        }
+
+        if (value < 0x100) { // uint 8
+          bytes.push(0xcc, value);
+          return 2;
+        }
+
+        if (value < 0x10000) { // uint 16
+          bytes.push(0xcd, value >> 8, value);
+          return 3;
+        }
+
+        if (value < 0x100000000) { // uint 32
+          bytes.push(0xce, value >> 24, value >> 16, value >> 8, value);
+          return 5;
+        }
+        // uint 64
+        hi = (value / Math.pow(2, 32)) >> 0;
+        lo = value >>> 0;
+        bytes.push(0xcf, hi >> 24, hi >> 16, hi >> 8, hi, lo >> 24, lo >> 16, lo >> 8, lo);
+        return 9;
+      } else {
+
+        if (value >= -0x20) { // negative fixnum
+          bytes.push(value);
+          return 1;
+        }
+
+        if (value >= -0x80) { // int 8
+          bytes.push(0xd0, value);
+          return 2;
+        }
+
+        if (value >= -0x8000) { // int 16
+          bytes.push(0xd1, value >> 8, value);
+          return 3;
+        }
+
+        if (value >= -0x80000000) { // int 32
+          bytes.push(0xd2, value >> 24, value >> 16, value >> 8, value);
+          return 5;
+        }
+        // int 64
+        hi = Math.floor(value / Math.pow(2, 32));
+        lo = value >>> 0;
+        bytes.push(0xd3, hi >> 24, hi >> 16, hi >> 8, hi, lo >> 24, lo >> 16, lo >> 8, lo);
+        return 9;
+      }
+      break;
+    case 'object':
+      // nil
+      if (value === null) {
+        bytes.push(0xc0);
+        return 1;
+      }
+
+      // uint 64
+      if (value._isUint64BE) {
+        value = value.toArray();
+        bytes.push(0xcf, value[0], value[1], value[2], value[3], value[4], value[5], value[6], value[7]);
+        return 9;
+      }
+
+      if (Array.isArray(value)) {
+        length = value.length;
+
+        if (length < 0x10) { // fixarray
+          bytes.push(length | 0x90);
+          size = 1;
+        } else if (length < 0x10000) { // array 16
+          bytes.push(0xdc, length >> 8, length);
+          size = 3;
+        } else if (length < 0x100000000) { // array 32
+          bytes.push(0xdd, length >> 24, length >> 16, length >> 8, length);
+          size = 5;
+        } else {
+          throw new Error('Array too large');
+        }
+        for (let i = 0; i < length; i++) {
+          size += _encode(bytes, defers, value[i]);
+        }
+        return size;
+      }
+
+      if (value instanceof Date) { // fixext 8 / Date
```

Comment:
We're never going to pass a date object in, so you might as well just not include this block. Leave a comment if you want indicating it was removed from the original.

Same goes for some of these other types.

Reply:
This check is quick enough that we can leave it in to stay as close as possible as the original code.